File: kotlin-study-repo/src/main/java/chap4/anonymous/AnonymousDemo.java
package chap4.anonymous;

import java.awt.event.MouseAdapter;
import java.awt.event.MouseEvent;

public class AnonymousDemo {
    public static void main(String[] args) {
        MouseAdapter mouseAdapter = new MouseAdapter() {
            @Override
            public void mouseClicked(MouseEvent e) {
                super.mouseClicked(e);
            }

            @Override
            public void mouseEntered(MouseEvent e) {
                super.mouseEntered(e);
            }
        };
    }
}

File: kotlin-study-repo/src/main/kotlin/chap4/anonymous/AnonymousDemo.kt
package chap4.anonymous

import java.awt.Window
import java.awt.event.MouseAdapter
import java.awt.event.MouseEvent

fun countClicks(window: Window) {
    var clickCount = 0

    window.addMouseListener(object : MouseAdapter() {
        override fun mouseClicked(e: MouseEvent) {
            clickCount++
        }
    })
}